server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +333,71 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const buildFacetForFilters = ({
+  isSexAtBirthFilterActive,
+  isInclusionCriteriaFilterActive,
+  userAccess,
+}) => {
+  const facetForFilters = {}
+
+  if (isSexAtBirthFilterActive) {
+    facetForFilters.socioDemographics = [
+      {
+        $match: {
+          assessment: SOCIODEMOGRAPHICS_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          socioDemographicsCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+  if (isInclusionCriteriaFilterActive) {
+    facetForFilters.inclusionCriteria = [
+      {
+        $match: {
+          assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+
+  return facetForFilters
+}
+
+export const intersectSubjectsFromFilters = (filters) => {
+  const intersectedSubjects = {}
+  const [filterListA, ...rest] = Object.values(filters).map((filter) => {
+    const filterMap = new Map()
+    filter.forEach((subjectData) => {
+      filterMap.set(subjectData.subject, { ...subjectData })
+    })
+
+    return filterMap
+  })
+  filterListA.forEach((subjectData) => {
```

Comment:
do we need to remove the subjects from `filterListA` that are not present in the other filters?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/cbd951fa9f62d42a3ae53518c2f1c4b48f02635b

Reply:
We'd have to remove them from the intersection Map/Object